src/SpectatorView.Unity/Assets/SpatialAlignment.ASA/Scripts/SpatialAlignment/SpatialAnchorsAndroidCoordinateService.cs

```diff
@@ -30,16 +31,31 @@ public SpatialAnchorsAndroidCoordinateService(SpatialAnchorsConfiguration spatia
         /// <inheritdoc/>
         protected override Task OnInitializeAsync()
         {
+            if (initialized)
```

Comment:
What I would recommend is to cache the taskCompletionSource in a static variable, and perform a null check on it to see if it needs instantiation or not. If it doesn't, just return it's task. Also, you can wrap all of it in a lock.